tests/tests/test_db_migration.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+# Copyright 2017 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+import pytest
+from common_setup import *
+from helpers import Helpers
+from MenderAPI import deploy
+from common_update import update_image_successful, common_update_procedure
+from mendertesting import MenderTesting
+
+class TestDBMigration(MenderTesting):
+
+    @pytest.mark.usefixtures("setup_with_legacy_client")
+    def test_migrate_from_legacy_mender_v1_failure(self, install_image=conftest.get_valid_image()):
+        """
+            Start a legacy client (1.7.0) first and update it to the new one.
+
+            The test starts a setup with the 1.7.0 client and then updates it to
+            the current version. The update is failing first (due to reboot
+            command inside the artifact commit enter state script).
```

Comment:
Where is this `reboot` actually happening? Because in the ArtifactCommit_Enter is only an `exit 1` but not reboot.